gramlib/grammar.ml

```diff
@@ -1614,10 +1614,13 @@ module Parsable = struct
       L.State.drop ();
       Exninfo.iraise (exn,info)
 
-  let make ?source cs =
+  let make ?loc cs =
     let lexer_state = ref (L.State.init ()) in
     L.State.set !lexer_state;
-    let ts = L.tok_func ?source cs in
+    (match loc with
+    | Some loc -> L.State.set_loc_offset Loc.(loc.bp)
+    | None -> ());
+    let ts = L.tok_func ?loc cs in
```

Comment:
Why do we need to set the offset twice here? Already `L.tok_func ?loc` should take care of that IIANM.